### PR TITLE
Merging vision into main

### PR DESCRIPTION
**Pull request by dirkgr**

### CODE REVIEW

**Comment by dirkgr** [comment]
The tests will now fail because we also need this on the -models side.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,10 +6,53 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 
-## Unreleased
+## Unreleased (2.x branch)
```
Comment (epwalsh) [inline CHANGELOG.md:9]:
```suggestion
## Unreleased
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,6 +65,46 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+### Fixed
+
+- The `build-vocab` command no longer crashes when the resulting vocab file is
+  in the current working directory.
+- VQA models now use the `vqa_score` metric for early stopping. This results in
+  much better scores.
+
+
+## Unreleased (1.x branch)
```
Comment (epwalsh) [inline CHANGELOG.md:95]: This section should probably be merged with above.

**Review by epwalsh — APPROVED** [review]
LSTM (looks scary to me), but let's do it

**Review by AkshitaB — APPROVED** [review]

**Comment by chenqiuyuan** [comment]
Hello, I notice that the parameter "cache_directory" is removed from the dataset_reader.py in this change. How can I use a similar configuration as "datasetreader.cache_directory": "my_cache_path"?

**Comment by dirkgr** [comment]
You can set the cache path by setting the environment variable `ALLENNLP_CACHE_ROOT`.

**Comment by dirkgr** [comment]
This was necessary because the cache is used in many different places, not just in dataset readers anymore.

**Comment by chenqiuyuan** [comment]
> 
> 
> You can set the cache path by setting the environment variable `ALLENNLP_CACHE_ROOT`.

Thank you very much : )!

**Pull request merged**